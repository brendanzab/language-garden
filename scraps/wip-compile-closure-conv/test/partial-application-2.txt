let add-captured : Int -> Int :=
  let x : Int := 5;
  let add-x : Int -> Int :=
    fun (y : Int) => x + y;

  add-x;

add-captured 3 + 4
