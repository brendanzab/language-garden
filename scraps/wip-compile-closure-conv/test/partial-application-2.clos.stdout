let add-captured : Int -> Int :=
  let x : Int := 5;
  let add-x : Int -> Int :=
    clos(fun (env : (Int,)) (y : Int) => #add env.0 y, (x,));
  add-x;
#add (add-captured 3) 4
